common/common/src/main/java/io/helidon/common/Version.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Simple class to provide version information.
+ */
+public class Version {
+
+    private static final String VERSION_PROPERTIES_FILE = "/version.properties";
+    private static final String UNKNOWN = "unknown";
```

Comment:
I'd avoid a default value as a String.
I'd catch `null` when initializing the values and throw an `IllegalStateException`.

Reply:
If we throw an exception, we may have problems in local development. I would agree if there is a way how to start web server from your IDE without the need to build using maven.

Reply:
Not sure why would would have a problem during development. When would the properties file not be in the class-path ?

Reply:
Wait are you able to start the webserver with all classes compiled on the fly by your IDE ??

Reply:
You are right. It would just print the unprocessed values

Reply:
> Wait are you able to start the webserver with all classes compiled on the fly by your IDE ??

yes

Reply:
And the IDE is clever enought to replace the properties:
`2019.03.22 21:21:33 INFO io.helidon.webserver.NettyWebServer !thread!: Helidon Version: 1.0.3-SNAPSHOT 2019-03-22T20:16:43Z`

Reply:
I'm confused. What am I right about ? If an exception is thrown nothing is printed right ?
Replace properties, you mean that your IDE processed the maven resources ?

Reply:
At least Idea processed the file correctly during IDE build, so for me this works fine.
Also the values would never be null if the property file is on the classpath. You may just get unprocessed `${project.version}` instead of the real version.
I think that throwing an exception would not really help, as that would be thrown when you attempt to run a web server. By which time we do not want to stop the user...

We should have a test that makes sure the values are replaced (e.g. you do not get "unknown" and you do not get `${project.version}`.

Reply:
Version class is now generated at build time. No more exceptions or runtime errors.